File: Server/src/main/java/StartRpcServer.java
import repository.*;
import server.*;
import services.IServices;

import java.io.IOException;
import java.util.Properties;

public class StartRpcServer {
    private static final int defaultPort = 55554;

    public static void main(String[] args) {
        Properties serverProps = new Properties();
        try {
            serverProps.load(StartRpcServer.class.getResourceAsStream("/server.properties"));
            System.out.println("Server properties set");
            serverProps.list(System.out);
        } catch (IOException ex) {
            System.err.println("Cannot find servet.properties: " + ex);
            return;
        }
        IAgencyRepo agencyRepo = new AgencyDBRepo(serverProps);
        ITripRepo tripRepo = new TripDBRepo(serverProps);
        IReservationRepo resRepo = new ReservationDBRepo(serverProps, tripRepo);
        AgencyService agencyService = new AgencyService(agencyRepo);
        TripService tripService = new TripService(tripRepo);
        ReservationService reservationService = new ReservationService(resRepo);

        IServices serverImpl = new Service(agencyService, tripService, reservationService);
        int serverPort = defaultPort;
        try {
            serverPort = Integer.parseInt(serverProps.getProperty("server.port"));
        } catch (NumberFormatException ex) {
            System.err.println("Wrong port number: " + ex.getMessage());
            System.err.println("Using default port: " + defaultPort);
        }
        System.out.println("Starting server on port: " + serverPort);
        AbstractServer server = new RpcConcurrentServer(serverPort, serverImpl);
        try {
            server.start();
        } catch (ServerException ex) {
            System.err.println("Error starting server: " + ex.getMessage());
        } finally {
            try {
                server.stop();
            } catch (ServerException ex) {
                System.err.println("Error stopping server: " + ex.getMessage());
            }
        }
    }
}

File: Server/StartRpcServer.cs
using System;
using System.Collections.Generic;
using System.Configuration;
using log4net.Config;
using Networking.server;
using Persistence;
using Server.server;

namespace Server
{
    public static class StartRpcServer
    {
        static void Main(string[] args)
        {
            XmlConfigurator.Configure(new System.IO.FileInfo("log4net.xml"));
            Console.WriteLine("Configuration Settings for agenciesDB {0}", GetConnectionStringByName("agenciesDB"));
            IDictionary<string, string> props = new SortedList<string, string>
            {
                { "ConnectionString", GetConnectionStringByName("agenciesDB") }
            };

            IAgencyRepo agencyRepo = new AgencyDBRepo(props);
            ITripRepo tripRepo = new TripDBRepo(props);
            IReservationRepo reservationRepo = new ReservationDBRepo(props, tripRepo);

            AgencyService agencyService = new(agencyRepo);
            TripService tripService = new(tripRepo);
            ReservationService reservationService = new(reservationRepo);

            Service service = new(agencyService, tripService, reservationService);

            //string host = "127.0.0.1";
            //int port = 55553;

            string host = ConfigurationManager.AppSettings.Get("host");
            int port = int.Parse(ConfigurationManager.AppSettings.Get("port"));
            Console.WriteLine("Port " + port);

            AbstractServer server = new RpcConcurrentServer(host, port, service);

            server.Start();
        }

        private static string GetConnectionStringByName(string name)
        {
            // Assume failure.
            string returnValue = null;

            // Look for the name in the connectionStrings section.
            ConnectionStringSettings settings = ConfigurationManager.ConnectionStrings[name];

            // If found, return the connection string.
            if (settings != null)
                returnValue = settings.ConnectionString;

            return returnValue;
        }
    }
}
